cpp/oneapi/dal/algo/kmeans/detail/train_ops.hpp

```diff
@@ -44,6 +44,12 @@ struct train_ops {
         if (!(input.get_data().has_data())) {
             throw domain_error(msg::input_data_is_empty());
         }
+        if (input.get_data().get_row_count() > dal::detail::limits<std::int32_t>::max()) {
+            throw domain_error(dal::detail::error_messages::row_count_gt_max_int32());
+        }
```

Comment:
Why do we add such precondition?

Reply:
Counters for clusters are int32_t to be efficient. It might cause overflow. There are two options: use int64 in kernel or check input for possible overflow. Last one looks more efficient for me.